### ISSUE TITLE
[Bug]: [benchmark][batch_insert]Frequent data insertion error: "deny to write"

### ISSUE THREAD

**Issue by elstic** [milvus-io/milvus#26257]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master-20230809-d267559b
- Deployment mode(standalone or cluster): both
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

argo task : fouramf-stable-1691611200 
case : test_concurrent_locust_diskann_compaction_cluster

case steps:  (it shows up on both diskann and hnsw indexes)
1. create a collection, build index, and insert 100,000 data;
2. flush collection 
3. build index 
4. load collection
5. perform concurrent operations (load、query、search、scene_insert_delete_flush)
"scene_insert_delete_flush" this scenario means, insert 3000 pieces of data, then delete, then flush_collection 

server:
```
fouramf-stable-11200-5-45-7414-etcd-0                             1/1     Running            0                 5h8m    10.104.15.213   4am-node20   <none>           <none>
fouramf-stable-11200-5-45-7414-etcd-1                             1/1     Running            0                 5h8m    10.104.6.23     4am-node13   <none>           <none>
fouramf-stable-11200-5-45-7414-etcd-2                             1/1     Running            0                 5h8m    10.104.20.22    4am-node22   <none>           <none>
fouramf-stable-11200-5-45-7414-milvus-datacoord-754b67fb94vfgpg   1/1     Running            0                 5h8m    10.104.19.224   4am-node28   <none>           <none>
fouramf-stable-11200-5-45-7414-milvus-datanode-64d8c77bfc-q6z2p   1/1     Running            1 (5h4m ago)      5h8m    10.104.4.71     4am-node11   <none>           <none>
fouramf-stable-11200-5-45-7414-milvus-indexcoord-67fccf9d4n67k6   1/1     Running            0                 5h8m    10.104.21.14    4am-node24   <none>           <none>
fouramf-stable-11200-5-45-7414-milvus-indexnode-5fdb4cf776qmkkz   1/1     Running            0                 5h8m    10.104.14.210   4am-node18   <none>           <none>
fouramf-stable-11200-5-45-7414-milvus-proxy-779f9c9c5f-jgfvm      1/1     Running            1 (5h4m ago)      5h8m    10.104.14.212   4am-node18   <none>           <none>
fouramf-stable-11200-5-45-7414-milvus-querycoord-b7f4d65d-6drbf   1/1     Running            1 (5h4m ago)      5h8m    10.104.14.209   4am-node18   <none>           <none>
fouramf-stable-11200-5-45-7414-milvus-querynode-5b5c9fb94bhfx42   1/1     Running            0                 5h8m    10.104.21.15    4am-node24   <none>           <none>
fouramf-stable-11200-5-45-7414-milvus-rootcoord-6d7bd946447hfcp   1/1     Running            1 (5h4m ago)      5h8m    10.104.14.206   4am-node18   <none>           <none>
fouramf-stable-11200-5-45-7414-minio-0                            1/1     Running            0                 5h8m    10.104.14.228   4am-node18   <none>           <none>
fouramf-stable-11200-5-45-7414-minio-1                            1/1     Running            0                 5h8m    10.104.12.216   4am-node17   <none>           <none>
fouramf-stable-11200-5-45-7414-minio-2                            1/1     Running            0                 5h8m    10.104.6.22     4am-node13   <none>           <none>
fouramf-stable-11200-5-45-7414-minio-3                            1/1     Running            0                 5h8m    10.104.20.18    4am-node22   <none>           <none>
fouramf-stable-11200-5-45-7414-pulsar-bookie-0                    1/1     Running            0                 5h8m    10.104.5.91     4am-node12   <none>           <none>
fouramf-stable-11200-5-45-7414-pulsar-bookie-1                    1/1     Running            0                 5h8m    10.104.15.220   4am-node20   <none>           <none>
fouramf-stable-11200-5-45-7414-pulsar-bookie-2                    1/1     Running            0                 5h8m    10.104.6.29     4am-node13   <none>           <none>
fouramf-stable-11200-5-45-7414-pulsar-bookie-init-6xx8g           0/1     Completed          0                 5h8m    10.104.6.10     4am-node13   <none>           <none>
fouramf-stable-11200-5-45-7414-pulsar-broker-0                    1/1     Running            0                 5h8m    10.104.18.90    4am-node25   <none>           <none>
fouramf-stable-11200-5-45-7414-pulsar-proxy-0                     1/1     Running            0                 5h8m    10.104.23.46    4am-node27   <none>           <none>
fouramf-stable-11200-5-45-7414-pulsar-pulsar-init-gthhp           0/1     Completed          0                 5h8m    10.104.6.9      4am-node13   <none>           <none>
fouramf-stable-11200-5-45-7414-pulsar-recovery-0                  1/1     Running            0                 5h8m    10.104.23.47    4am-node27   <none>           <none>
fouramf-stable-11200-5-45-7414-pulsar-zookeeper-0                 1/1     Running            0                 5h8m    10.104.12.217   4am-node17   <none>           <none>
fouramf-stable-11200-5-45-7414-pulsar-zookeeper-1                 1/1     Running            0                 5h6m    10.104.16.233   4am-node21   <none>           <none>
fouramf-stable-11200-5-45-7414-pulsar-zookeeper-2                 1/1     Running            0                 5h4m    10.104.21.38    4am-node24   <none>   
```

client log:
```
[2023-08-09 20:51:49,906 - ERROR - fouram]: RPC error: [batch_insert], <MilvusException: (code=53, message=deny to write, reason: memory quota exhausted, please allocate more resources, req: /milvus.proto.milvus.MilvusService/Insert)>, <Time:{'RPC start': '2023-08-09 20:51:49.803584', 'RPC error': '2023-08-09 20:51:49.906449'}> (decorators.py:108)
[2023-08-09 20:51:49,906 - ERROR - fouram]: (api_response) : <MilvusException: (code=53, message=deny to write, reason: memory quota exhausted, please allocate more resources, req: /milvus.proto.milvus.MilvusService/Insert)> (api_request.py:53)
[2023-08-09 20:51:49,906 - ERROR - fouram]: [CheckFunc] insert request check failed, response:<MilvusException: (code=53, message=deny to write, reason: memory quota exhausted, please allocate more resources, req: /milvus.proto.milvus.MilvusService/Insert)> (func_check.py:52)
[2023-08-09 20:51:49,907 - ERROR - fouram]: [func_time_catch] :  (api_request.py:120)
[2023-08-09 20:51:50,433 - ERROR - fouram]: RPC error: [batch_insert], <MilvusException: (code=53, message=deny to write, reason: memory quota exhausted, please allocate more resources, req: /milvus.proto.milvus.MilvusService/Insert)>, <Time:{'RPC start': '2023-08-09 20:51:50.426846', 'RPC error': '2023-08-09 20:51:50.433711'}> (decorators.py:108)
[2023-08-09 20:51:50,434 - ERROR - fouram]: (api_response) : <MilvusException: (code=53, message=deny to write, reason: memory quota exhausted, please allocate more resources, req: /milvus.proto.milvus.MilvusService/Insert)> (api_request.py:53)
[2023-08-09 20:51:50,434 - ERROR - fouram]: [CheckFunc] insert request check failed, response:<MilvusException: (code=53, message=deny to write, reason: memory quota exhausted, please allocate more resources, req: /milvus.proto.milvus.MilvusService/Insert)> (func_check.py:52)
[2023-08-09 20:51:50,434 - ERROR - fouram]: [func_time_catch] :  (api_request.py:120)
[2023-08-09 20:51:51,717 - ERROR - fouram]: RPC error: [batch_insert], <MilvusException: (code=53, message=deny to write, reason: memory quota exhausted, please allocate more resources, req: /milvus.proto.milvus.MilvusService/Insert)>, <Time:{'RPC start': '2023-08-09 20:51:51.699690', 'RPC error': '2023-08-09 20:51:51.717505'}> (decorators.py:108)
[2023-08-09 20:51:51,718 - ERROR - fouram]: (api_response) : <MilvusException: (code=53, message=deny to write, reason: memory quota exhausted, please allocate more resources, req: /milvus.proto.milvus.MilvusService/Insert)> (api_request.py:53)
[2023-08-09 20:51:51,718 - ERROR - fouram]: [CheckFunc] insert request check failed, response:<MilvusException: (code=53, message=deny to write, reason: memory quota exhausted, please allocate more resources, req: /milvus.proto.milvus.MilvusService/Insert)> (func_check.py:52)
[2023-08-09 20:51:51,719 - ERROR - fouram]: [func_time_catch] :  (api_request.py:120)
```

client test result:
Problem: scene_insert_delete_flush scene fails 90% of the time

memory usage:
queryndoe resource: limit:8g,request 5g
Monitoring the querynode memory usage is close to the limit, probably related to the write ban.
![image](https://github.com/milvus-io/milvus/assets/48523564/4deb5ae1-97e9-48e0-9f94-58dbde0d33e4)

### Expected Behavior

No write prohibition will occur

When the case passes, the querynode memory usage is only 1.3G.
![image](https://github.com/milvus-io/milvus/assets/48523564/403861bd-be2b-4c37-a364-5fd9fc7e267f)

### Steps To Reproduce

```markdown
1. create a collection or use an existing collection
        2. build index on vector column
        3. insert a certain number of vectors
        4. flush collection
        5. build index on vector column with the same parameters
        6. build index on on scalars column or not
        7. count the total number of rows
        8. load collection
        9. perform concurrent operations
        10. clean all collections 
```

#### How to reproduce: 
</b></b>
Go to the gtp platform and run the case: `test_concurrent_locust_diskann_compaction_cluster`

</b></b></b></b>
</b></b>
#### An identifiable point in time?
</b></b>
 It's recent, it was normal at the time of this image : master-20230805-241117dd

**Comment by yanliang567** [comment]
@elstic please add more info about by which rule it triggers the write deny?
/assign @elstic

**Comment by elstic** [comment]
> @elstic please add more info about by which rule it triggers the write deny? /assign @elstic

okay, I added that memory usage monitoring might be useful

**Comment by elstic** [comment]
My guess is that the deletion of the data didn't take effect, causing the querynode memory to only go up, not down; need further confirmation from the devs

**Comment by xiaofan-luan** [comment]
what do you mean by "Frequent insertion of datagrams"?

**Comment by elstic** [comment]
> what do you mean by "Frequent insertion of datagrams"?

I have a problem with the presentation here, I've changed the title

**Comment by XuanYang-cn** [comment]
After index&compaction, there're only 340 segments, but querynode's memory is already fully loaded 2.3k+800 segment, has  no space left to replace the new, larger segments.
![image](https://github.com/milvus-io/milvus/assets/51370125/38206a22-a634-49c7-827e-eaace057c0c5)
![image](https://github.com/milvus-io/milvus/assets/51370125/2269e607-04b7-4bac-9111-ce031c2431b7)

**Comment by XuanYang-cn** [comment]
if insert 3000 and then delete 3000, the querynode should remove the segment when compaction done and reduce the memeory. 
Still investigating.

**Comment by XuanYang-cn** [comment]
Seems very like cp stucked

**Comment by XuanYang-cn** [comment]
perhaps could be fixed by https://github.com/milvus-io/milvus/pull/26534
/assign @elstic

**Comment by elstic** [comment]
The issue has been fixed, verify the image : master-20230823-148446cf
